File: 1598-crawler-log-folder/solution.cc
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int depth = 0;
        for (auto& l : logs) {
            if (l[0] == '.' && l[1] == '.') {
                if (depth > 0) --depth;
            } else if (l[0] == '.' && l[1] == '/') {
                continue;
            } else {
                ++depth;
            }
        }
        return depth;
    }
};

File: 1598-crawler-log-folder/solution.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        depth = 0
        for log in logs:
            stride = log.split('/')
            for d in stride:
                if len(d) == 0: continue
                if d == '..':
                    if depth > 0:
                        depth -= 1
                elif d != '.':
                    depth += 1
        return depth

